tsctl

```diff
@@ -616,28 +309,136 @@ class SearchTemplateManager(Command):
                 db_session.commit()
 
 
-if __name__ == '__main__':
+class ImportTimeline(Command):
+    """Create a new Timesketch timeline from a file."""
+    option_list = (
+        Option('--file', '-f', dest='file_path', required=True),
+        Option('--sketch_id', '-s', dest='sketch_id', required=False),
+        Option('--username', '-u', dest='username', required=False),
+        Option('--timeline_name', '-n', dest='timeline_name',
+               required=False),
+    )
+
+    def __init__(self):
+        super(ImportTimeline, self).__init__()
+
+    # pylint: disable=arguments-differ, method-hidden
+    def run(self, file_path, sketch_id, username, timeline_name):
+        """This is the run method."""
+
+        file_path = os.path.realpath(file_path)
+        file_path_no_extension, extension = os.path.splitext(file_path)
+        extension = extension.lstrip('.')
+        filename = os.path.basename(file_path_no_extension)
+
+        if not os.path.isfile(file_path):
+            sys.exit('No such file: {0:s}'.format(file_path))
+
+        if extension not in ('plaso', 'csv', 'jsonl'):
+            sys.exit('Unknown extension: {0:s}'.format(file_path))
+
+        user = None
+        if not username:
+            username = pwd.getpwuid(os.stat(file_path).st_uid).pw_name
+        if username is not 'root':
+            user = User.query.filter_by(username=unicode(username)).first()
+        if not user:
+            sys.exit('Cannot determine user for file: {0:s}'.format(file_path))
+
+        sketch = None
+        # If filename starts with <number> then use that as sketch_id.
+        # E.g: 42_file_name.plaso means sketch_id is 42.
+        sketch_id_from_filename = filename.split('_')[0]
+        if not sketch_id and sketch_id_from_filename.isdigit():
+            sketch_id = sketch_id_from_filename
+
+        if sketch_id:
+            try:
+                sketch = Sketch.query.get_with_acl(sketch_id, user=user)
+            except Forbidden:
+                pass
+
+        if not timeline_name:
+            timeline_name = unicode(filename.replace('_', ' '))
+            # Remove sketch ID if present in the filename.
+            if timeline_name.split()[0].isdigit():
```

Comment:
can you make this more readable by splitting it into two lines, not have the .split() and [] and isdigit all in one line

Reply:
Done.